src/components/TransactionBlock/PlrDaoStakingTransactionBlock.tsx

```diff
@@ -0,0 +1,632 @@
+    resetTransactionBlockFieldValidationError(transactionBlockId, 'amount');
+    resetTransactionBlockFieldValidationError(transactionBlockId, 'offer');
+    resetTransactionBlockFieldValidationError(
+      transactionBlockId,
+      'toAssetAddress'
+    );
+  }, [selectedFromNetwork]);
+
+  const getWalletBalance = async (chainId: number, name: string) => {
+    const accountBalancePromise = [];
+    if (accountAddress && providerAddress) {
+      accountBalancePromise.push(getSupportedAssetsWithBalancesForChainId(chainId, true, accountAddress));
+      accountBalancePromise.push(getSupportedAssetsWithBalancesForChainId(chainId, true, providerAddress));
+    }
+
+    return await Promise.allSettled(accountBalancePromise)
+      .then((response: any[]) => {
+        const accountWalletBalance = response.map(accountBalance => accountBalance?.status == 'fulfilled' && accountBalance?.value);
+
+        var smartWalletBalance = 0;
+        var smartWalletUSD = 0;
+        
+        var keyBasedBalance = 0;
+        var keyBasedUSD = 0;
+
+        accountWalletBalance[0]?.forEach(({ symbol, decimals, balance, balanceWorthUsd}) => {
+          if(symbol == "PLR") {
+            smartWalletBalance += +ethers.utils.formatUnits(balance, decimals);
+          } else {
+            smartWalletUSD += balanceWorthUsd;
+          }
+        });
+        accountWalletBalance[1]?.forEach(({ symbol, decimals, balance, balanceWorthUsd}) => {
+          if(symbol == "PLR") {
+            keyBasedBalance += +ethers.utils.formatUnits(balance, decimals);
+          } else {
+            keyBasedUSD += balanceWorthUsd;
+          }
+        });
+
+        return {
+          chainName: name,
+          keyBasedWallet: keyBasedBalance,
+          smartWallet: smartWalletBalance,
+          keyBasedUSD: keyBasedUSD,
+          smartWalletUSD: smartWalletUSD,
+        };
+      })
+      .catch((error) => {
+        return {
+          chainName: name,
+          keyBasedWallet: 0,
+          smartWallet: 0,
+          keyBasedUSD: 0,
+          smartWalletUSD: 0,
+        }
+      });
+  };
+
+  const getBalanceForAllChains = async () => {
+    // TODO:UNCOMMENT
+    const chainPromise = [];
+    chainPromise.push(getWalletBalance(supportedChains[1].chainId, supportedChains[1].title))
+    // chainPromise.push(getWalletBalance(supportedChains[2].chainId, supportedChains[2].title))
+    chainPromise.push(getWalletBalance(supportedChains[3].chainId, supportedChains[3].title))
+    // chainPromise.push(getWalletBalance(supportedChains[3].chainId, supportedChains[3].title))
+    // supportedChains.forEach(async (chain) => {
+    //   chainPromise.push(getWalletBalance(chain.chainId, chain.title))
+    // });
+    return await Promise.allSettled(chainPromise).catch((e) => {
+      return [];
+    });
+  }
+
+  const fetchAllAccountBalances = async () => {
+    var accountBalanceWithSupportedChains = await getBalanceForAllChains().then((data) => data.map((d) => d.value));
```

Comment:
Remember to `.catch` here just incase

Reply:
if you ever use `await` please do not mix if with `then...catch`! even though it works it is considered as mixing 2 types of promise resolving (await wrapper with try...catch VS then...catch)

first wait for your method, then do mapping

```suggestion
    const data = await getBalanceForAllChains();
    const accountBalanceWithSupportedChains = data.map((d) => d.value));
```